modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java

```diff
@@ -1332,18 +1332,35 @@ public List<Long> getPrimaryKeys(
 				objectDefinitionId
 			).and(
 				() -> {
-					if (groupId == 0) {
+					if (ArrayUtil.isEmpty(groupIds)) {
 						return null;
 					}
 
-					return ObjectEntryTable.INSTANCE.groupId.eq(groupId);
+					return ObjectEntryTable.INSTANCE.groupId.in(groupIds);
 				}
 			).and(
 				Predicate.withParentheses(
 					_fillPredicate(objectDefinitionId, predicate, search))
 			).and(
-				_getPermissionWherePredicate(
-					dynamicObjectDefinitionTable, groupId)
+				() -> {
+					Predicate permissionWherePredicate = null;
+
+					for (Long groupId : groupIds) {
+						if (permissionWherePredicate == null) {
+							permissionWherePredicate =
+								_getPermissionWherePredicate(
+									dynamicObjectDefinitionTable, groupId);
+						}
+						else {
+							permissionWherePredicate =
+								permissionWherePredicate.or(
+									_getPermissionWherePredicate(
+										dynamicObjectDefinitionTable, groupId));
+						}
+					}
+
+					return permissionWherePredicate;
+				}
```

Comment:
can you isolate this piece of code also? Or maybe include it inside _getPermissionWherePredicate? One of these two options, up to you